src/core.c/Main.pm6

```diff
@@ -72,21 +73,31 @@ my sub RUN-MAIN(&main, $mainline, :$in-as-argsfiles) {
 
             # named
             elsif $passed-value
-              ~~ /^ [ '--' | '-' | ':' ] ('/'?) (<-[0..9\.]> .*) $/ {  # 'hlfix
-                my str $arg = $1.Str;
+              ~~ /^ ( '--' | '-' | ':' ) ('/'?) (<-[0..9\.]> .*) $/ {  # 'hlfix
+                my str $arg = $2.Str;
                 my $split  := nqp::split("=",$arg);
 
                 # explicit value
                 if nqp::isgt_i(nqp::elems($split),1) {
                     my str $name = nqp::shift($split);
-                    %named.push: $name => $0.chars
+                    die "Can't combine bundling with explicit arguments" if $bundling && nqp::iseq_s($0, '-');
+                    %named.push: $name => $1.chars
                       ?? thevalue(nqp::join("=",$split)) but False
                       !! thevalue(nqp::join("=",$split));
                 }
 
                 # implicit value
                 else {
-                    %named.push: $arg => !($0.chars);
+                    if $bundling && nqp::iseq_s($0.Str, '-') {
+                        die "Can't combine bundling with explicit negation" if $1.chars;
+                        my @chars = nqp::split('',$arg);
+                        for @chars -> $char {
```

Comment:
Couldn't we just do:

    for nqp::split('',$arg) -> $char {

??

Reply:
Sure

Reply:
Actually we can't, that dies with «Cannot find method 'iterator' on 'BOOTArray': no method cache and no .^find_method»